### PR TITLE
Runs tests on travis-ci + Enforces use of MathJax2 for math rendering in Docs

### PR DESCRIPTION
**Pull request by navidcy**
This PR updates `.travis.yml` to run all tests + build the documentation.

Furthermore, it enforces the use of MathJax2 for rendering maths in the docs thus fixing the current rendering issues.

Last, it drops some unnecessary dependencies like Plots.jl.

### CODE REVIEW

**Comment by navidcy** [comment]
I couldn't see Plots being used anywhere in the main package.

(I did see some `use_pyplot_utils` macro which was calling PyPlot... but PyPlot was not a dependency so I don't know how that works...)

**Comment by glwagner** [comment]
Thanks so much @navidcy .

There are no plots in the documentation; thus no need for PyPlot. The macro is just a convenience for users who do have `PyPlot` in their environment. It works because it is a macro; therefore that code is not actually part of the package. I do think it should be nuked at some point in favor of Plots everywhere.

File: docs/make.jl [hunk 1]
```diff
@@ -7,7 +7,8 @@ makedocs(
        clean = true,
      doctest = false,
    checkdocs = :all,
-      format = Documenter.HTML(prettyurls = get(ENV, "CI", nothing) == "true"),
+      format = Documenter.HTML(prettyurls = get(ENV, "CI", nothing) == "true",
+                               mathengine = Documenter.MathJax2()),
```
Comment (glwagner) [inline docs/make.jl:15]: <3

**Review by glwagner — APPROVED** [review]

**Comment by navidcy** [comment]
OK, I figured out why travis was failing. TKEMassFlux tests were taking too long. I did a hack to avoid travis timeout.

Apparently the `.travis.yml` config file **before** this PR was only running the docs! That's why it was passing no probs....

**Comment by navidcy** [comment]
I can't merge so you'll have to do it ;)

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,22 @@
 language: julia
-julia: 1.1
-os: linux
```
Comment (navidcy) [inline .travis.yml:3]: this is not the right way to ask travis to run tests and, therefore, travis was only building the docs

Comment (glwagner) [reply .travis.yml:3]: you mean, the syntax was wrong? thanks for fixing it.

Comment (navidcy) [reply .travis.yml:3]: Yes that’s what I meant 🤙🏼

**Comment by glwagner** [comment]
> OK, I figured out why travis was failing. TKEMassFlux tests were taking too long. I did a hack to avoid travis timeout.
> 
> Apparently the `.travis.yml` config file **before** this PR was only running the docs! That's why it was passing no probs....

 Awesome, thanks a lot @navidcy .

**Pull request merged**